### ISSUE TITLE
Rubocop issues a Style/GuardClause, when the clause is the entire method body.

### ISSUE THREAD

**Issue by saveriomiroddi** [rubocop-hq/rubocop#5702]
When a method consists of a single clause, Rubocop raises an error, expecting a guard clause pattern to be applied.

There is no code to guard though (there would be if there was an `else` block), so rubocop is forcing the if/raise to be split in two.

Note that in this case, the if and its content can't be converted to a single line, since they would exceed the allowed length.

--------

## Expected behavior

Rubocop should not issue an offense.

## Actual behavior

Rubocop issues an offense.

## Steps to reproduce the problem

This is a simple test case:

```ruby
# The raise statement almost reaches the allowed legnth.
def check_sources_consistency!(options)
  if options[:git] && options[:'local-git']
    raise InvalidOption, "Remote and local plugin git sources can't be..."
  end
end
```

The single-line guard form can't be used, due to line length, forcing the method to be rewritten as:

```ruby
def code_as_wanted_by_rubocop(options)
  return unless options[:git] && options[:'local-git']

  raise InvalidOption, "Remote and local plugin git sources can't be..."
end
```

This forces the introduction of artificial code to guard (the `raise` statement), while on the original method, there is no code/logic to guard.

## RuboCop version

```
0.53.0 (using Parser 2.5.0.4, running on ruby 2.4.3 x86_64-linux)
```

**Comment by pocke** [comment]
Probably RuboCop expects below.

```ruby
def code_as_wanted_by_rubocop(options)
  if options[:git] && options[:'local-git']
    raise InvalidOption, "Remote and local plugin git sources can't be..."
  end

  puts 'This is the regular but nonexistent code...'
  puts 'Run! Bundler is now taking over your machine.'
end
```

**Comment by Drenmi** [comment]
This would also work:

```ruby
def code_as_wanted_by_rubocop(options)
  raise InvalidOption, 'Ooops.' if options[:git] && options[:'local-git']
end
```

I'm not sure how to assess the valuations "not proper" and "weird". All three options seem perfectly readable to me. 🙂

**Comment by saveriomiroddi** [comment]
> I'm not sure how to assess the valuations "not proper" and "weird". All three options seem perfectly readable to me.

I'll update the description to make it clearer.

The first exception is very typical code, but it's refused by Rubocop.

The second example is the way I think Rubocop interprets the first example. It's also typical code, and in that case, it would be appropriate to use a guard clause.

The third form is acceptable by Rubocop, and that's OK. I think it's strange to force such code, since it disconnects the error raise from the condition it triggers.

**Comment by mikegee** [comment]
> What I found improper is that it forces the introduction of artificial code to guard (the raise statement), while on the original method, there is no code/logic to guard.

I don't interpret that example method as you have described. When I write "assertion" methods like your example I prefer writing `return unless condition; raise "the error"`. I don't think there's anything "artificial" about it. Perhaps I've been brainwashed by Rubocop, though 🤔
